test/colors/hsl_test.rb

```diff
@@ -0,0 +1,269 @@
+require_relative '../test_helper'
+
+class ColorsHSLTest < Test::Unit::TestCase
+  include TestHelper
+
+  sub_test_case(".new") do
+    test("with integer values") do
+      c = Charty::Colors::HSL.new(1, 128, 255)
+      assert_equal(1r, c.hue)
+      assert_equal(128/255r, c.saturation)
+      assert_equal(255/255r, c.lightness)
+
+      c = Charty::Colors::HSL.new(-1, 128, 255)
+      assert_equal(359r, c.hue)
+
+      c = Charty::Colors::HSL.new(361, 128, 255)
+      assert_equal(1r, c.hue)
+
+      assert_raise(ArgumentError) do
+        Charty::Colors::HSL.new(0, 0x100, 0)
+      end
+
+      assert_raise(ArgumentError) do
+        Charty::Colors::HSL.new(0, 0, 0x100)
+      end
+
+      assert_raise(ArgumentError) do
+        Charty::Colors::HSL.new(0, -1, 0)
+      end
+
+      assert_raise(ArgumentError) do
+        Charty::Colors::HSL.new(0, 0, -1)
+      end
```

Comment:
It's better that we split a test that has many assertions to small tests that have only a few assertions.

Reply:
OK.  I filed the new issue #43.

I don't want to do it now because I don't want to consume a lot of time to consider test descriptions.